tests/dns_mock.rs

```diff
@@ -11,133 +13,37 @@ use trust_dns_client::udp::UdpClientConnection;
-                        0x65,
-                        0x03,
-                        0x63,
-                        0x6f,
-                        0x6d,
-                        0x00,
-                        0x00,
-                        0x01,
-                        0x00,
-                        0x01,
-                        0xc0,
-                        0x0c,
-                        0x00,
-                        0x01,
-                        0x00,
-                        0x01,
-                        0x00,
-                        0x01,
-                        0x08,
-                        0xa4,
-                        0x00,
-                        0x04,
-                        0x5d,
-                        0xb8,
-                        0xd8,
-                        0x22,
-                        0xc0,
-                        0x10,
-                        0x00,
-                        0x02,
-                        0x00,
-                        0x01,
-                        0x00,
-                        0x01,
-                        0x08,
-                        0xa3,
-                        0x00,
-                        0x14,
-                        0x01,
-                        0x61,
-                        0x0c,
-                        0x69,
-                        0x61,
-                        0x6e,
-                        0x61,
-                        0x2d,
-                        0x73,
-                        0x65,
-                        0x72,
-                        0x76,
-                        0x65,
-                        0x72,
-                        0x73,
-                        0x03,
-                        0x6e,
-                        0x65,
-                        0x74,
-                        0x00,
-                        0xc0,
-                        0x10,
-                        0x00,
-                        0x02,
-                        0x00,
-                        0x01,
-                        0x00,
-                        0x01,
-                        0x08,
-                        0xa3,
-                        0x00,
-                        0x04,
-                        0x01,
-                        0x62,
-                        0xc0,
-                        0x3f,
-                        0x00,
-                        0x00,
-                        0x29,
-                        0x10,
-                        0x00,
-                        0x00,
-                        0x00,
-                        0x00,
-                        0x00,
-                        0x00,
-                        0x00,
-                    ])
-                },
-            ),
+            SendMessageDependingOnLastReceivedMessage(|previous_message| {
+                let mut response = vec![
+                    previous_message.as_ref().unwrap()[0],
+                    previous_message.as_ref().unwrap()[1],
+                ];
+                response.extend_from_slice(&[
```

Comment:
Thanks, the previous formatting has been generated by rustfmt, which is sometimes quite strange

Reply:
fighting rustfmt is a fun challenge :)